### ISSUE TITLE
build: allow to specify Lua executable

### ISSUE THREAD

**Issue by blueyed** [awesomeWM/awesome#3196]
The "awesome" package in Arch Linux uses (https://github.com/archlinux/svntogit-community/blob/32a597fe3e1ddcb50eec45a6e57cc4f5a9d20b1a/trunk/PKGBUILD#L32-L35):
```bash
  sed -i 's/COMMAND lua /COMMAND lua5.3 /' awesomeConfig.cmake
  sed -i 's| lua | lua5.3 |;s/lua)/lua5.3)/' tests/examples/CMakeLists.txt
  sed -i '1s|/usr/bin/env lua$|/usr/bin/env lua5.3|' build-utils/check_for_invalid_requires.lua
  sed -i 's/"lua"/"lua5.3"/' tests/test-spawn.lua
```

The "awesome-luajit-git" package (maintained by me, and initially based on the "awesome" PKGBUILD) uses by now:
```
prepare() {
  cd $pkgname
  sed -i 's/COMMAND lua\b/COMMAND luajit/' awesomeConfig.cmake tests/examples/CMakeLists.txt
  sed -i 's/LUA_COV_RUNNER lua\b/LUA_COV_RUNNER luajit/' tests/examples/CMakeLists.txt
}
```

To not require packagers to manually update/fix this it would be good if there was some CMake build variable to specify the Lua interpreter / binary to use.

For reference, this is what Neovim uses: https://github.com/neovim/neovim/blob/a260d5def363ca30f0473365ea514f2dedd0d0ba/CMakeLists.txt#L483-L503

**Comment by psychon** [comment]
Whoops, sorry.

**Comment by unai-ndz** [comment]
Related: #1377, #1053.

**Comment by sclu1034** [comment]
This was implemented in #3279.
